src/hooks/useNotes.ts

```diff
@@ -0,0 +1,78 @@
+import { Wallet } from "ethers";
+import { useCallback, useEffect, useState } from "react";
+
+import { NoteItemProps } from "../components/NoteItem/NoteItem";
+import { useGlobalState } from "../contexts/global";
+import { getFeedUpdate } from "../utils/bee";
+import { ADDRESS_HEX_LENGTH, SELF_NOTE_TOPIC } from "../utils/constants";
+import { getLocalPrivateKey } from "../utils/helpers";
+
+export const useNotes = () => {
+  const { notes, setNotes } = useGlobalState();
+  const [noteRawTopics, setNoteRawTopics] = useState<string[]>([]);
+
+  const fetchNotes = useCallback(async () => {
+    const privKey = getLocalPrivateKey();
+    if (!privKey) {
+      console.error("private key not found - cannot fetch notes");
+      return;
+    }
+
+    const wallet = new Wallet(privKey); // TODO: remove wallet and use PrivateKey everywhere
+    const feedPromises: Promise<string>[] = [];
+    for (let i = 0; i < noteRawTopics.length; i++) {
+      const rawTopic = noteRawTopics[i];
+      feedPromises.push(getFeedUpdate(wallet.address, rawTopic));
+    }
+
+    const notesArray: string[] = [];
+    const results = await Promise.allSettled(feedPromises);
+    results.forEach((result) => {
+      if (result.status === "fulfilled") {
+        if (result.value.length > 0) {
+          notesArray.push(result.value);
+        }
+      } else {
+        console.error(`fetching note data error: `, result.reason);
+      }
+    });
+
+    const tmpNotes: NoteItemProps[] = [...notes];
+    for (let i = 0; i < notesArray.length; i++) {
+      let note: NoteItemProps | undefined = undefined;
+      try {
+        note = JSON.parse(notesArray[i]) as NoteItemProps;
+      } catch (error) {
+        console.error(`error parsing notes[${i}]:\n ${error}`);
+        continue;
+      }
+      const found = notes.find((n) => n.id === note.id);
```

Comment:
nitpick: If the found element is not used, then `notes.some()` could be used and get the boolean value.